File: урок2/football.js
function bestTeamSum(players) {
    let bestSum = 0;
    let nowSum = 0;
    let last = 0;

    for (let first = 0; first < players.length; first++) {
        while (last < players.length && (last === first || players[first] + players[first + 1] >= players[last])) {
            nowSum += players[last];
            last++;
        }
        bestSum = Math.max(bestSum, nowSum);
        nowSum -= players[first];
    }

    return bestSum;
}

console.log(bestTeamSum([1, 2, 3, 5, 6, 8, 9]));

File: урок2/football.py
def bestteamsum(players):
    bestsum = 0
    nowsum = 0
    last = 0
    for first in range(len(players)):
        while last < len(players) and (last == first or players[first] + players[first + 1] >= players[last]):
            nowsum +=players[last]
            last += 1
        bestsum = max(bestsum, nowsum)
        nowsum -= players[first]
    return bestsum
print(bestteamsum([14, 15, 17, 21, 30]))